webkitgtk@2.4.11 build failure

Starting with an empty `/usr/local` on macOS 10.12.6 (16G29), I first installed Homebrew, then added the astroid tap, and then tried to install astroid. This gave me the following compile errors in webkitgtk:
```
==> Installing astroidmail/astroid/astroid dependency: webkitgtk@2.4.11
==> Downloading https://webkitgtk.org/releases/webkitgtk-2.4.11.tar.xz
Already downloaded: /Users/alexanderadolf/Library/Caches/Homebrew/webkitgtk@2.4.11-2.4.11.tar.xz
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/webkitgtk@2.4.11/2.4.11 --disable-introspection --disable-x11-target --enable-q
==> make
Last 15 lines from /Users/alexanderadolf/Library/Logs/Homebrew/webkitgtk@2.4.11/02.make:
Source/JavaScriptCore/API/JSStringRef.cpp:65:35: error: no matching function for call to 'createWithoutCopying'
    return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./Source/WTF/wtf/text/StringImpl.h:393:54: note: candidate function not viable: no known conversion from 'const JSChar *' (aka 'const unsigned short *') to 'const UChar *' (aka 'const char16_t *') for 1st argument
    WTF_EXPORT_STRING_API static PassRef<StringImpl> createWithoutCopying(const UChar* characters, unsigned length);
                                                     ^
./Source/WTF/wtf/text/StringImpl.h:394:54: note: candidate function not viable: no known conversion from 'const JSChar *' (aka 'const unsigned short *') to 'const LChar *' (aka 'const unsigned char *') for 1st argument
    WTF_EXPORT_STRING_API static PassRef<StringImpl> createWithoutCopying(const LChar* characters, unsigned length);
                                                     ^
Source/JavaScriptCore/API/JSStringRef.cpp:86:12: error: cannot initialize return object of type 'const JSChar *' (aka 'const unsigned short *') with an rvalue of type 'const UChar *' (aka 'const char16_t *')
    return string->characters();
           ^~~~~~~~~~~~~~~~~~~~
3 errors generated.
make[1]: *** [Source/JavaScriptCore/API/libjavascriptcoregtk_3_0_la-JSStringRef.lo] Error 1
make: *** [all] Error 2
```
It would seem all of them are rooted in no cast/conversion between `JSChar` and `UChar` being available?

There was a [discussion on NetBSD back in April](https://mail-index.netbsd.org/pkgsrc-users/2017/04/23/msg024731.html) on a seemingly very similar issue, in which someone seems to have found a solution. Not sure if their fix would solve this too, and whether or why it was/wasn't fixed upstream though.

--- a/Formula/webkitgtk@2.4.11.rb
+++ b/Formula/webkitgtk@2.4.11.rb
@@ -13,10 +13,13 @@ class WebkitgtkAT2411 < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
 
+  patch :p0, :DATA
+
   def install
     ENV.deparallelize
 
     icu4c = Formula["icu4c"]
+
 
     # build options without ./configure flags
     inreplace "Source/autotools/SetupWebKitFeatures.m4",
@@ -72,3 +75,69 @@ class WebkitgtkAT2411 < Formula
     system "false"
   end
 end
+
+__END__
+------------------------------------------------------------------------
+r216187 | annulen@yandex.ru | 2017-05-05 00:33:41 +0900 (Fri, 05 May 2017) | 28 lines
+
+Fix compilation with ICU 59.1
+https://bugs.webkit.org/show_bug.cgi?id=171612
+
+Reviewed by Mark Lam.
+
+ICU 59.1 has broken source compatibility. Now it defines UChar as
+char16_t, which does not allow automatic type conversion from unsigned
+short in C++ code.
+
+--- Source/JavaScriptCore/API/JSStringRef.cpp.orig  2016-04-10 06:48:36 UTC
++++ Source/JavaScriptCore/API/JSStringRef.cpp
+@@ -37,7 +37,7 @@ using namespace WTF::Unicode;
+ JSStringRef JSStringCreateWithCharacters(const JSChar* chars, size_t numChars)
+ {
+     initializeThreading();
+-    return OpaqueJSString::create(chars, numChars).leakRef();
++    return OpaqueJSString::create(reinterpret_cast<const UChar*>(chars), numChars).leakRef();
+ }
+
+ JSStringRef JSStringCreateWithUTF8CString(const char* string)
+@@ -62,7 +62,7 @@ JSStringRef JSStringCreateWithUTF8CString(const char*
+ JSStringRef JSStringCreateWithCharactersNoCopy(const JSChar* chars, size_t numChars)
+ {
+     initializeThreading();
+-    return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
++    return OpaqueJSString::create(StringImpl::createWithoutCopying(reinterpret_cast<const UChar*>(chars), numChars)).leakRef();
+ }
+
+ JSStringRef JSStringRetain(JSStringRef string)
+@@ -83,7 +83,7 @@ size_t JSStringGetLength(JSStringRef string)
+
+ const JSChar* JSStringGetCharactersPtr(JSStringRef string)
+ {
+-    return string->characters();
++    return reinterpret_cast<const JSChar*>(string->characters());
+ }
+
+ size_t JSStringGetMaximumUTF8CStringSize(JSStringRef string)
+--- Source/JavaScriptCore/runtime/DateConversion.cpp.orig 2013-08-03 16:10:38 UTC
++++ Source/JavaScriptCore/runtime/DateConversion.cpp
+@@ -107,7 +107,8 @@ String formatDateTime(const GregorianDateTime& t, Date
+ #if OS(WINDOWS)
+             TIME_ZONE_INFORMATION timeZoneInformation;
+             GetTimeZoneInformation(&timeZoneInformation);
+-            const WCHAR* timeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
++            const WCHAR* winTimeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
++            String timeZoneName(reinterpret_cast<const UChar*>(winTimeZoneName));
+ #else
+             struct tm gtm = t;
+             char timeZoneName[70];
+--- Source/WebKit2/Shared/API/c/WKString.cpp.orig 2016-04-10 06:48:37 UTC
++++ Source/WebKit2/Shared/API/c/WKString.cpp
+@@ -55,7 +55,7 @@ size_t WKStringGetLength(WKStringRef stringRef)
+ size_t WKStringGetCharacters(WKStringRef stringRef, WKChar* buffer, size_t bufferLength)
+ {
+     COMPILE_ASSERT(sizeof(WKChar) == sizeof(UChar), WKStringGetCharacters_sizeof_WKChar_matches_UChar);
+-    return (toImpl(stringRef)->getCharacters(static_cast<UChar*>(buffer), bufferLength));
++    return (toImpl(stringRef)->getCharacters(reinterpret_cast<UChar*>(buffer), bufferLength));
+ }
+
+ size_t WKStringGetMaximumUTF8CStringSize(WKStringRef stringRef)

